cmd/copyS2SMigrationBlobEnumerator.go

```diff
@@ -260,6 +266,13 @@ func (e *copyS2SMigrationBlobEnumerator) addBlobToNTransfer(srcURL, destURL url.
 
 func (e *copyS2SMigrationBlobEnumerator) addBlobToNTransfer2(srcURL, destURL url.URL, properties *azblob.BlobGetPropertiesResponse,
 	cca *cookedCopyCmdArgs) error {
+	util := copyHandlerUtil{}
+
+	bt := properties.BlobType()
+	if e.BlobAttributes.BlobType == common.EBlobType.None() || e.BlobAttributes.BlobType == common.EBlobType.Detect() {
+		bt = util.inferBlobType(srcURL.Path, bt)
```

Comment:
Note that there's not need to specifically address this comment, other than running the test, if you do the `getTargetBlobType` approach that I suggested in my overall end-of-review comment.

Reply:
Your answer is: We already lie to the system.

Check `ste/s2sCopier-URLToBlob.go:43`. We manually override from STE. This scenario is no different in implementation.

Reply:
Also, please take a look at the rest of my comment above. I'm fairly sure this PR is introducing a bug, and we can't ship with that.  Details in original comment above.  And that bug is another key reason why we should do everything in the STE. Since it's the easiest way to fix that bug. The only other way to fix that bug is to introduce a new field in the plan files (as you raised in another comment) to store _desired destination blob type_ in a new field, and keep the existing _source blob type_ as is.

Reply:
Correction: it looks like there's code in line 194 of sender-pageBlob that will actually prevent the bug from triggering. So even though in line 43 of sender-pageBlobFromURL we will actually pick up an invalid tier name, line 194 of sender-pageBlob will prevent it from being used.  So there won't be any problem at runtime.  I just confused me when reading the source, since I didn't realise we had _two_ levels of protection against the problem.

So there is no tier preservation bug.  Sorry for the false alarm.

Reply:
Ah, no worries.